src/webapp/components/survey-list/hook/useDeleteSurvey.ts

```diff
@@ -0,0 +1,40 @@
+import { useState } from "react";
+import { SURVEY_FORM_TYPES } from "../../../../domain/entities/Survey";
+import { useCurrentSurveys } from "../../../contexts/current-surveys-context";
+
+import { useAppContext } from "../../../contexts/app-context";
+import i18n from "@eyeseetea/feedback-component/locales";
+import { Id } from "../../../../domain/entities/Ref";
+
+export interface DeleteState {
+    status: "success" | "error";
+    message: string;
+}
+
+export function useDeleteSurvey(formType: SURVEY_FORM_TYPES, refreshSurveys: any) {
```

Comment:
the beauty of using custom hooks is, that you can reuse it in multiple places. Instead of passing refreshSurveys from SurveyList (3 levels above) , you can get refreshSurveys from useSurveys  inside  useDeleteSurvey hook. What you have done is not wrong, do you think this suggestion would make is simpler?

Reply:
@mariaozamiz As discussed , I made a mistake here. Each each call to a customHook has a completely isolated state , so we need to pass the refreshSurveys from the parent component as you had done earlier. Apologies! Here is a reference to the react documentation explaining this: https://legacy.reactjs.org/docs/hooks-overview.html#:~:text=In%20fact%2C%20each%20call%20to,it%20is%20a%20custom%20Hook.